printf: More test and doc cleanup (#8290)
KhronosGroup/Vulkan-ValidationLayers: docs/debug_printf.md

Debug Printf settings can also be managed using the [Vulkan Configurator](https://vulkan.lunarg.com/doc/sdk/latest/windows/vkconfig.html) included with the Vulkan SDK.

### Settings

> All settings are found in Vulkan Configurator (`VkConfig`)

> There are currently 3 environmental variables that are used for settings in Debug PrintF

Reviewer: the uppercase F in PrintF is strange...

> There are currently 3 environmental variables that are used for settings in Debug PrintF

Reviewer 2: ```suggestion
There are currently 3 environment variables that are used for settings in Debug PrintF
```

- `VK_LAYER_PRINTF_TO_STDOUT` will print to `stdout` **instead** of the normal Debug Callback
    - `VK_LAYER_PRINTF_TO_STDOUT=1` turn on
    - `VK_LAYER_PRINTF_TO_STDOUT=0` turn off

> - `VK_LAYER_PRINTF_VERBOSE` will print extra information which pipeline, shader, command, etc, the printf message came from

Reviewer: does not read well / is not syntactically correct

    - `VK_LAYER_PRINTF_VERBOSE=1` turn on
    - `VK_LAYER_PRINTF_VERBOSE=0` turn off

> - `VK_LAYER_PRINTF_BUFFER_SIZE` size of buffer to allocate to put printf message too (this is shared across all calls in a single `vkQueueSubmit`)

Reviewer: `too` => `to` or `into`

> - `VK_LAYER_PRINTF_BUFFER_SIZE` size of buffer to allocate to put printf message too (this is shared across all calls in a single `vkQueueSubmit`)

Reviewer: plural: messages

> - `VK_LAYER_PRINTF_BUFFER_SIZE` size of buffer to allocate to put printf message too (this is shared across all calls in a single `vkQueueSubmit`)

Reviewer 2: ```suggestion
- `VK_LAYER_PRINTF_BUFFER_SIZE` size of the buffer used to store Printf messages (buffer is shared across all calls in a single `vkQueueSubmit`). Default: 1024 bytes.
```

>     - `VK_LAYER_PRINTF_BUFFER_SIZE=1024` default size

Reviewer 2: ```suggestion
```

#version 450
#extension GL_EXT_debug_printf : enable
void main() {
    float myfloat = 3.1415f;
    debugPrintfEXT("My float is %f", myfloat);
}
```
Then use `glslangValidator` to generate SPIR-V to use in `vkCreateShaderModule`.
`glslangvalidator --target-env vulkan1.2 -e main -o shader.vert.spv shader.vert` would be one
example of compiling shader.vert

Note that every time this shader is executed, `My float is 3.141500` will be printed. If this were
in a vertex shader and a triangle was drawn, it would be printed 3 times.

> Note also that the `VK_KHR_shader_non_semantic_info` device extension must be enabled in

Reviewer 2: This info seems quite important, but lost at the end of the paragraph, I think it should be elsewhere

Reviewer 2: It is also repeated in the `Using Debug Printf in SPIR-V Shaders` section

Author: will remove the duplicate, but I need to look into this, not full sure this is needed as a requirement since we implement it in the shader instrumentation

Documentation for the GL_EXT_debug_printf extension can be found
[here](https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_debug_printf.txt)

> There are many validation layer test that demonstrates the simple and programmatic use of Debug

Reviewer: plural: `many [...] tests [...] that demonstrate`
